Algorithm.CSharp/TimeInForceAlgorithm.cs

```diff
@@ -70,6 +73,15 @@ public override void OnData(Slice data)
                 DefaultOrderProperties.TimeInForce = TimeInForce.Day;
                 _dayOrderTicket = LimitOrder(_symbol, 10, 160m);
             }
+
+            if (_gtdOrderTicket == null)
+            {
+                // This order will expire on October 10th at market close,
+                // if not filled by then it will be canceled automatically.
+
+                DefaultOrderProperties.TimeInForce = new GoodTilDateTimeInForce(new DateTime(2013, 10, 10));
```

Comment:
I'm wondering if it makes more sense to define `GoodTilDateTimeInForce` using a `TimeSpan` instead of an actual `DateTime`. Using a time span I think is more natural and also doesn't require it to be updated every day. Also, from an implementation standpoint it's easily supported since the order object has the submit time on it, so can easily compute the date we should cancel. Thoughts?

Reply:
Also, since this is under regression test, maybe we should have a case where GTD is cancelled _and_ GTD gets executed. Also, let's confirm that other TIF have both sides exercised in the regression algorithm, otherwise we're only testing half of the implementation via regression.

Reply:
Another reason here, I'm thinking of eventually supporting syntax like:
``` csharp
DefaultOrderProperties.TimeInForce = TimeSpan.FromDays(10);
```
via an implicit operator which would require the data held to be a time span value instead of an absolute date time value.

Reply:
The only issue we would have changing `GTD` from `DateTime Expiry` to `TimeSpan Duration` is that not all brokerage API have the order submission time available, so we would not be able to get back the `Duration` value in `GetOpenOrders`.

IB and FXCM do not have order creation/submission time in their get order API methods while OANDA and GDAX have this field.

IB, FXCM and OANDA have the expiry date/time field, GDAX doesn't have it but in its `GTT` time in force only allows fixed durations of 1 minute, 1 hour or 1 day.

Maybe we need to have both properties?